mne/viz/topo.py

```diff
@@ -72,88 +72,319 @@ def iter_topography(info, layout=None, on_pick=None, fig=None,
         The related channel index.
     """
     import matplotlib.pyplot as plt
+    from ..channels.layout import find_layout
+
+    if pos is None:
+        if layout is None:
+            layout = find_layout(info)
+        pos = layout.pos.copy()
+        if layout_scale is not None:
+            pos[:, :2] *= layout_scale
 
     if fig is None:
         fig = plt.figure()
 
-    fig.set_facecolor(fig_facecolor)
-    if layout is None:
-        from ..channels import find_layout
-        layout = find_layout(info)
-
     if on_pick is not None:
         callback = partial(_plot_topo_onpick, show_func=on_pick)
         fig.canvas.mpl_connect('button_press_event', callback)
 
+    fig.set_facecolor(fig_facecolor)
+
+    ch_names = _clean_names(info['ch_names'])
+    ch_types = dict((ch_name, channel_type(info, j))
+                    for j, ch_name in enumerate(ch_names))
+
+    iter_ch = [(x, y) for x, y in enumerate(info["ch_names"]) if y in ch_names]
+    for idx, ch_name in iter_ch:
+        if ch_name in find_layout(info).names:
+            ax = plt.axes(pos[idx])
```

Comment:
There's no guarantee that the index of a channel in the Epochs object is the same as the index of the channel in the layout object. You want the latter:

    ax = plt.axes(pos[layout.names.index(ch_name)])